rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -414,10 +458,21 @@ impl PendingMessage {
         pm
     }
 
-    fn on_reprepare(&mut self) -> PendingOperationResult {
+    fn on_reprepare<E: Debug>(
+        &mut self,
+        err: E,
+        reason: ReprepareReason,
+    ) -> PendingOperationResult {
         self.inc_attempts();
         self.submitted = false;
-        PendingOperationResult::Reprepare
+        warn!(error = ?err, "{}", reason.clone());
```

Comment:
I assume info about the PendingMessage will be in some span that's shown in this log?

Reply:
not right now, but I'll include this